clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -878,8 +879,30 @@ void EmitAssemblyHelper::EmitAssembly(BackendAction Action,
   PerFunctionPasses.add(
       createTargetTransformInfoWrapperPass(getTargetIRAnalysis()));
 
+  if (LangOpts.SYCLIsDevice && LangOpts.SYCLExplicitSIMD) {
+    PerModulePasses.add(createSYCLLowerESIMDPass());
+  }
+
   CreatePasses(PerModulePasses, PerFunctionPasses);
 
+  if (LangOpts.SYCLIsDevice && LangOpts.SYCLExplicitSIMD &&
```

Comment:
I can't find review for this part, so I'll ask here.

Origianal code uses `CreatePasses` to populate `PerModulePasses` pass manager. It seems reasonable to move new code adding ESIMD passes to this function. Right?

Reply:
Good point, will do.